babel/messages/extract.py

```diff
@@ -137,10 +138,11 @@ def extract_from_dir(dirname=None, method_map=DEFAULT_MAPPING,
 
     absname = os.path.abspath(dirname)
     for root, dirnames, filenames in os.walk(absname):
-        dirnames[:] = [
-            subdir for subdir in dirnames
-            if not (subdir.startswith('.') or subdir.startswith('_'))
-        ]
+        for subdir in dirnames:
+            if ignore_dir_prefixes:
```

Comment:
If `ignore_dir_prefixes` defaults to an empty tuple, this check is unnecessary. Also, it should be done outside the `for subdir...` loop anyway, as there's no way the prefix list will change during that iteration.